lib/peeps.rb

```diff
@@ -0,0 +1,96 @@
+require 'pg'
+require 'time'
+require './lib/users.rb'
+require './lib/tags.rb'
+
+class Peeps
+
+  attr_reader :id, :user_id, :message, :time, :user,  :replying_to_id
+
+  def initialize(id, user_id, message, time, replying_to_id, user=Users.user_with_id(user_id))
+    p "the id is #{id}"
+    p "the userid is #{user_id}"
+    @id = id.to_i
+    @user_id = user_id.to_i
+    @message = message
+    @time = time
+    @user = user
+    @replying_to_id = replying_to_id
+  end
+
+  def self.all
+    peeps = sql('
+                      SELECT p.id, user_id, message, time, name, username
+                      FROM peeps p
+                      left join users u
+                      on u.id = user_id
+                      order by time desc;')
+    peeps.map { |peep| Peeps.new(peep['id'],
+                                peep['user_id'],
+                                peep['message'],
+                                Time.parse(peep['time']),
+                                peep['name'],
+                                peep['username']
+                              )
+                      }
+  end
+
+
+  def self.create(user_id, peep_content, replying_to="NULL")
+    new_peeps = sql("insert into peeps(user_id, message, replying_to_id)
+          values (#{user_id}, '#{peep_content}', #{replying_to})
+          RETURNING id, user_id, message, time, replying_to_id;")
+          new_peep = new_peeps[0]
+          new_peep = Peeps.new(new_peep["id"],
+                    new_peep["user_id"],
+                    new_peep["message"],
+                    new_peep["time"],
+                    new_peep["replying_to_id"]
+                    )
+          new_peep.update_tags
+          new_peep
+  end
+
+  def update_tags
+    tags = self.message.scan(/@\w+/)
+    tags.each do |tag|
+      p tag
```

Comment:
remove p statements. You can leave the ones that will help in debugging the server later on, like "#{username} just signed in", but not ones used for debugging while coding.